### DISCUSSION TITLE
For a Purchase, what IS the transactionId property? It seems to be constantly changing.

### DISCUSSION THREAD

**Discussion by ProtegraChris** [hyochan/react-native-iap#2674] [Q&A]
I'm on iOS and testing subscriptions in sandbox mode.

When I get my list of purchases via getPurchaseHistory, everything comes back fine as I'd expect. If I close and reopen the app and get my purchases again in exactly the same way, the data still comes back as expected, but all the transactionIds are different from what they were the previous time. This continues every time they're fetched.

Unless this is just a quirk of the sandbox environment and transactionId is consistent in normal environments, what is this property actually tracking that's changing so often?

**Comment by AdamGerthel** [comment]
I'm wondering about this as well. Also noticed that `transactionReceipt` is an optional value - but when does it not appear? I've noticed that *some* of our users that have received an IAP through code redemption do not have a `transactionReceipt` value when the purchase is restored - but that's the only time that seems to happen - and even then it's not always missing.
